src/open-uri.c

```diff
@@ -664,6 +664,15 @@ handle_open_in_thread_func (GTask *task,
       g_autoptr(GError) local_error = NULL;
 
       path = xdp_app_info_get_path_for_fd (request->app_info, fd, 0, NULL, &fd_is_writable, &local_error);
+
+      char *host_path = get_real_path_for_doc_path (path, request->app_info);
+      if (host_path)
+        {
+          g_debug ("OpenFile: translating path value '%s' to host path '%s'", path, host_path);
+          g_free(path);
+          path = host_path;
+        }
+
```

Comment:
I think using `g_autofree` would be nice here. Something like this:

```c
      g_autoptr(GError) local_error = NULL;
      g_autofree char *host_path = NULL;

      path = xdp_app_info_get_path_for_fd (request->app_info, fd, 0, NULL, &fd_is_writable, &local_error);

      host_path = get_real_path_for_doc_path (path, request->app_info);
      if (host_path)
        {
          g_debug ("OpenFile: translating path value '%s' to host path '%s'", path, host_path);
          g_clear_pointer (&path, g_free);
          path = g_steal_pointer (&host_path);
        }
```

(In the future we'll be able to use `g_set_str()` here, which will be even nicer, but not now because we'd be depending on a GLib version that's too recent)